Sprint 09

KatSchreiner opened the pull request.

KatSchreiner committed:
Реализован профиль пользователя ProfileViewController

KatSchreiner committed:
Реализован переход с таблицы к SingleImageView

KatSchreiner committed:
Реализован 'scrollView' для 'SingleImageView'. Добавлен 'zoom' и 'scroll' к 'image'

KatSchreiner committed:
Свёрстан кодом профиля пользователя

KatSchreiner committed:
доработки

So1ovyov commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -35,6 +36,16 @@ final class ImagesListViewController: UIViewController {
         tableView.dataSource = self
         }
```
Поехал отступ, давай поправим

So1ovyov commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,88 @@
+final class ProfileViewController: UIViewController {
+    
+    // MARK: - Private Properties
+    private var userPhoto = UIImageView()
```
Предлагаю разгрузить методы ниже и сразу при создании элемента настраивать его: 
```
    private var userPhoto: UIImageView = {
        let image = UIImage(named: "Photo")
        let view = UIImageView(image: image)
        view.translatesAutoresizingMaskIntoConstraints = false
        return view
    }()
```

So1ovyov commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,88 @@
+        let imageProfile = UIImage(named: "Photo")
+        userPhoto = UIImageView(image: imageProfile)
+        userPhoto.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(userPhoto)
```
Обрати внимание на инструмент `forEach` при верстке и настройке элементов, он поможет тебе сократить повторяющийся код
Вот пример:
```
        [view1, 
         view2,
         view3].forEach {
            $0.translatesAutoresizingMaskIntoConstraints = false
             view.addSubview($0)
        }
```
Таким образом все строки `translatesAutoresizingMaskIntoConstraints` и `addSubview` выше можно удалить и очень сильно сократить код

So1ovyov commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,88 @@
+        userPhoto = UIImageView(image: imageProfile)
+        userPhoto.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(userPhoto)
+        userPhoto.widthAnchor.constraint(equalToConstant: 70).isActive = true
```
Для того, чтобы каждого свойство не активировать `.isActive = true`, предлагаю использовать инструмент `NSLayoutConstraint.activate[]`
```
        NSLayoutConstraint.activate([
            userPhoto.widthAnchor.constraint(equalToConstant: 70),
            userPhoto.heightAnchor.constraint(equalToConstant: 70),
            userPhoto.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 16),
            userPhoto.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 30)
        ])
```

KatSchreiner committed:
Внесены изменения:
- Поправила отступы
- Исправила констрейнты для ScrollView
- Сократила повторяющийся код translatesAutoresizingMaskIntoConstraints и addSubview
- Использовала инструмент NSLayoutConstraint.activate[]
- Разгрузила методы ProfileViewController
